tools-local/tasks/BuildFPMToolPreReqs.Netstandard2_0.cs

```diff
@@ -0,0 +1,581 @@
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// This task prepares the command line parameters for running a RPM build using FPM tool and also updates the copyright and changelog file tokens.
+    /// If parses various values from the config json by first reading it into a model and then builds the required string for parameters and passes it back.
+    /// 
+    /// </summary>
+    public class BuildFPMToolPreReqs : BuildTask
+    {
+        [Required]
+        public string InputDir { get; set; }
+        [Required]
+        public string OutputDir { get; set; }
+        [Required]
+        public string PackageVersion { get; set; }
+        [Required]
+        public string ConfigJsonFile { get; set; }
+        [Output]
+        public string FPMParameters { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                if (!File.Exists(ConfigJsonFile))
+                {
+                    throw new FileNotFoundException($"Expected file {ConfigJsonFile} was not found.");
+                }
+
+                // Open the Config Json and read the values into the model
+                TextReader projectFileReader = File.OpenText(ConfigJsonFile);
+                if (projectFileReader != null)
+                {
+                    string jsonFileText = projectFileReader.ReadToEnd();
+
+                    // TODO: Replace the use of the Utf8JsonReader with the Deserializer once it is available.
+                    ConfigJson configJson = GetConfigJson(jsonFileText);
+
+                    // Update the Changelog and Copyright files by replacing tokens with values from config json
+                    UpdateChangelog(configJson, PackageVersion);
+                    UpdateCopyRight(configJson);
+
+                    // Build the full list of parameters 
+                    FPMParameters = BuildCmdParameters(configJson, PackageVersion);
+                    Log.LogMessage(MessageImportance.Normal, "Generated RPM paramters:  " + FPMParameters);
+                }
+                else
+                {
+                    throw new IOException($"Could not open the file {ConfigJsonFile} for reading.");
+                }
+            }
+            catch (Exception e)
+            {
+                Log.LogError("Exception while processing RPM paramters: " + e.Message);
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private ConfigJson GetConfigJson(string jsonFileText)
+        {
+            byte[] utf8Json = Encoding.UTF8.GetBytes(jsonFileText);
+            var configJson = new ConfigJson();
+
+            var reader = new Utf8JsonReader(utf8Json, isFinalBlock: true, state: default);
+
+            while (reader.Read() && reader.TokenType == JsonTokenType.PropertyName)
+            {
+                string propertyName = reader.GetStringValue().ToLower();
+                if (propertyName == Maintainer_Name_String)
```

Comment:
Does perf not matter here?  Or could we instead compare span to span rather than materializing the string?  And if we stick with strings, could the comparison be done as ordinal ignore case rather than lower casing the property name?